lib/modules/manager/gomod/update.ts

```diff
@@ -17,16 +17,17 @@ export function updateDependency({
 }: UpdateDependencyConfig): string | null {
   try {
     logger.debug(`gomod.updateDependency: ${upgrade.newValue}`);
-    const { depName, depType, updateType } = upgrade;
+    const { depType, updateType } = upgrade;
+    const fromPackageName = upgrade.depName;
```

Comment:
I have a little concern about readability here. packageName is a formal concept in Renovate, separate from depName. This use of packageName in fromPackageName is not really in line with how we use the packageName concept. Does a term like "currentDepName" fit instead?

Regarding your replacement one, do you think it would be semantically wrong to do something like `let depName = upgrade.depName` and then reassign it later if needed?

Reply:
or use `fromDepName`?

Reply:
I was sure that I'd copied this pattern from somewhere else in the codebase, but I now can't see it, so I must have been mistaken.

+1 to the use of `packageName` not being the same as what we're doing here, and happy to update it to `fromDepName`.

In terms of:

>  do you think it would be semantically wrong to do something like `let depName = upgrade.depName` and then reassign it later if needed?

My intent was to make it possible to surface the to/from data so you're aware i.e. in the logs that it's a replacement.

However, that's something we're not doing elsewhere, and so I'm happy to replace with a `let` :+1: 

In which case, I believe this PR is no longer needed, and we can do it all in #29575?